tools/importer-msgraph-metadata/components/parser/resourceids.go

```diff
@@ -62,6 +62,16 @@ func (ri ResourceIds) MatchIdOrAncestor(resourceId ResourceId) (*ResourceIdMatch
 	return nil, false
 }
 
+// ResourceId represents a unique path in Microsoft Graph that represents a resource. Resource IDs comprise a non-zero
+// number of segments, the last of which must always of type SegmentUserValue (i.e. specified by the user).
+// Whilst paths in Microsoft Graph can comprise different types of object identifiers, we currently only support simple
+// SegmentUserValue segments where the entire slug is provided. For example, these two paths are functionally equivalent:
+//
+//	/applications/{id}/federatedIdentityCredentials/{federatedIdentityCredentialId}
+//	/applications(appId='{appId}')/federatedIdentityCredentials/{federatedIdentityCredentialName}
```

Comment:
I'm not familiar with this notation at all, but is there a `/` missing here?
```suggestion
//	/applications/(appId='{appId}')/federatedIdentityCredentials/{federatedIdentityCredentialName}
```

Reply:
Alas there is not... this is largely why we don't/can't currently support this. These complex segments essentially represent _two_ segments in SDK terms, and there's also the problem of them being partly user-specified and partly static, so we'd need to rework our generated ID parsers to accommodate these.